test/e2e/e2e_toomanyrestarts_test.go

```diff
@@ -0,0 +1,188 @@
+						Ports:           []v1.ContainerPort{{ContainerPort: 80}},
+					}},
+				},
+			},
+		},
+	}
+	_, err = clientSet.AppsV1().Deployments(deploymentObj.Namespace).Create(ctx, deploymentObj, metav1.CreateOptions{})
+	if err != nil {
+		t.Logf("Error creating deployment: %v", err)
+		if err = clientSet.AppsV1().Deployments(deploymentObj.Namespace).DeleteCollection(ctx, metav1.DeleteOptions{}, metav1.ListOptions{
+			LabelSelector: labels.SelectorFromSet(labels.Set(map[string]string{"test": "restart-pod", "name": "test-toomanyrestarts"})).String(),
+		}); err != nil {
+			t.Fatalf("Unable to delete deployment: %v", err)
+		}
+		return
+	}
+	defer clientSet.AppsV1().Deployments(deploymentObj.Namespace).Delete(ctx, deploymentObj.Name, metav1.DeleteOptions{})
+
+	// Need to wait restartCount more than 4
+	result, err := waitPodRestartCount(ctx, clientSet, testNamespace.Name, t)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+	if !result {
+		t.Fatal("Pod restart count not as expected")
+	}
+
+	tests := []struct {
+		name                    string
+		podRestartThreshold     int32
+		includingInitContainers bool
+		expectedEvictedPodCount int
+	}{
+		{
+			name:                    "test-no-evictions",
+			podRestartThreshold:     int32(10000),
+			includingInitContainers: true,
+			expectedEvictedPodCount: 0,
+		},
+		{
+			name:                    "test-one-evictions",
+			podRestartThreshold:     int32(4),
+			includingInitContainers: true,
+			expectedEvictedPodCount: 4,
+		},
+	}
+	for _, tc := range tests {
+		evictionPolicyGroupVersion, err := eutils.SupportEviction(clientSet)
+		if err != nil || len(evictionPolicyGroupVersion) == 0 {
+			t.Fatalf("Error creating eviction policy group: %v", err)
+		}
+		podEvictor := evictions.NewPodEvictor(
+			clientSet,
+			evictionPolicyGroupVersion,
+			false,
+			0,
+			nodes,
+			true,
+			false,
+			false,
+		)
+		// Run RemovePodsHavingTooManyRestarts strategy
+		t.Log("Running RemovePodsHavingTooManyRestarts strategy")
+		strategies.RemovePodsHavingTooManyRestarts(
+			ctx,
+			clientSet,
+			deschedulerapi.DeschedulerStrategy{
+				Enabled: true,
+				Params: &deschedulerapi.StrategyParameters{
+					PodsHavingTooManyRestarts: &deschedulerapi.PodsHavingTooManyRestarts{
+						PodRestartThreshold:     tc.podRestartThreshold,
+						IncludingInitContainers: tc.includingInitContainers,
+					},
+				},
+			},
+			workerNodes,
+			podEvictor,
+		)
+
+		waitForTerminatingPodsToDisappear(ctx, t, clientSet, testNamespace.Name)
+		actualEvictedPodCount := podEvictor.TotalEvicted()
+		if actualEvictedPodCount < tc.expectedEvictedPodCount {
+			t.Errorf("Fail to run test case %v, actual evicted pod number is: %d", tc.name, actualEvictedPodCount)
+		}
+	}
+}
+
+func waitPodRestartCount(ctx context.Context, clientSet clientset.Interface, namespace string, t *testing.T) (bool, error) {
+	timeout := time.After(5 * time.Minute)
+	tick := time.Tick(5 * time.Second)
+	for {
```

Comment:
Could you use `wait.PollImmediate` instead of for+select? `wait.PollImmediate` will give you the same functionality, just a bit cleaner.

Reply:
Hi. I agree with you. At first I use `wait.PollImmediate`, but soon I give up.
That's for I need to execute List pods many times and util all pod restart count more than 4.
But I I use `wait.PollImmediate`, I have to return when I execute the first time to List Pods.
So I can use `for` for I need to execute list pods many times.